pkg/dependency/credhelper/cred_helper_binary.go

```diff
@@ -0,0 +1,189 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package credhelper
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/opencontainers/go-digest"
+	"github.com/spf13/afero"
+
+	"github.com/runfinch/finch/pkg/command"
+	"github.com/runfinch/finch/pkg/config"
+	"github.com/runfinch/finch/pkg/dependency"
+	"github.com/runfinch/finch/pkg/flog"
+	"github.com/runfinch/finch/pkg/path"
+
+	"github.com/docker/cli/cli/config/configfile"
+)
+
+type credhelperbin struct {
+	fp         path.Finch
+	fs         afero.Fs
+	cmdCreator command.Creator
+	l          flog.Logger
+	fc         *config.Finch
+	user       string
+	hcfg       helperConfig
+}
+
+var _ dependency.Dependency = (*credhelperbin)(nil)
+
+func newCredHelperBinary(fp path.Finch, fs afero.Fs, cmdCreator command.Creator, l flog.Logger, fc *config.Finch,
+	user string, hcfg helperConfig,
+) *credhelperbin {
+	return &credhelperbin{
+		// TODO: consider replacing fp with only the strings that are used instead of the entire type
+		fp:         fp,
+		fs:         fs,
+		cmdCreator: cmdCreator,
+		l:          l,
+		fc:         fc,
+		user:       user,
+		hcfg:       hcfg,
+	}
+}
+
+func updateConfigFile(bin *credhelperbin) error {
+	cfgPath := fmt.Sprintf("%s%s", bin.hcfg.finchPath, "config.json")
+	binCfgName := bin.credHelperConfigName()
+	fileExists, err := afero.Exists(bin.fs, cfgPath)
+	if err != nil {
+		return err
+	}
+	if !fileExists {
+		file, err := bin.fs.Create(cfgPath)
+		if err != nil {
+			return err
+		}
+		JSONstr := fmt.Sprintf("{\"credsStore\":\"%s\"}", binCfgName)
```

Comment:
I would assume it's out of scope for this PR, but might be worth opening an issue.

See https://github.com/awslabs/amazon-ecr-credential-helper/issues/474#issuecomment-1538822507

Users can also configure credential helper as

```
{
	"credHelpers": {
		"public.ecr.aws": "ecr-login",
		"<aws_account_id>.dkr.ecr.<region>.amazonaws.com": "ecr-login"
	}
}
```

Reply:
since this PR has been updated, the direction appears to be supporting the syntax for `"credHelpers"` rather than `"credsStore"`. Should we be updating how `config.json` sets credential helpers, then? As of right now in this PR, a user could provide multiple credential helpers in `finch.yaml`, but only one of those would be added to the config

Reply:
Since this will require additional user input in the form of `<aws_account_id>` and `<region>` I think it might be better to leave this responsibility to the user. Other wise additional fields need to be added to finch.yaml so the user can configure it.